### PR TITLE
Add Api.TypesSpec test cases for SealedTx request body

### PR DESCRIPTION
**Pull request by sevanspowell**
### Description

This PR contains some suggestions/extra work for #2896.

### Ticket

ADP-919

### CODE REVIEW

File: lib/core/test/unit/Cardano/Wallet/Api/Malformed.hs [hunk 1]
```diff
@@ -1513,7 +1515,12 @@ instance Malformed (BodyParam (ApiBytesT 'Base64 SerialisedTx))
 -- in Server.hs. Tested by integration tests.
 
 instance Malformed (BodyParam (ApiT SealedTx)) where
-    malformed = [] -- fixme: add a couple things
+    malformed = first BodyParam <$>
+        [ ("", "DecoderErrorDeserialiseFailure 'Shelley Tx' (DeserialiseFailure 0 'end of input')")
+        , ("hello", "DecoderErrorDeserialiseFailure 'Shelley Tx' (DeserialiseFailure 0 'expected list len or indef')")
+        , ("cafecafe", "DecoderErrorDeserialiseFailure 'Shelley Tx' (DeserialiseFailure 0 'expected list len or indef')")
+        , (BSL.fromStrict . T.encodeUtf8 $ validSealedTxBase64, "")
```
Comment (sevanspowell) [inline lib/core/test/unit/Cardano/Wallet/Api/Malformed.hs]: I would expect the case of using a "validSealedTxBase64" to succeed, but it does not. Fails with:

```
DecoderErrorDeserialiseFailure 'Shelley Tx' (DeserialiseFailure 0 'expected list len or indef')
```

File: lib/core/test/unit/Cardano/Wallet/Api/Malformed.hs [hunk 2]
```diff
@@ -1513,7 +1515,12 @@ instance Malformed (BodyParam (ApiBytesT 'Base64 SerialisedTx))
 -- in Server.hs. Tested by integration tests.
 
 instance Malformed (BodyParam (ApiT SealedTx)) where
-    malformed = [] -- fixme: add a couple things
+    malformed = first BodyParam <$>
+        [ ("", "DecoderErrorDeserialiseFailure 'Shelley Tx' (DeserialiseFailure 0 'end of input')")
```
Comment (sevanspowell) [inline lib/core/test/unit/Cardano/Wallet/Api/Malformed.hs:1547]: Are these the error messages you would expect @rvl?

File: lib/core/test/unit/Cardano/Wallet/Api/Malformed.hs (hunk 1 above)
Comment (sevanspowell) [reply lib/core/test/unit/Cardano/Wallet/Api/Malformed.hs]: Perhaps I'm converting it to ByteString incorrectly.

Comment (rvl) [reply lib/core/test/unit/Cardano/Wallet/Api/Malformed.hs]: Oops, this endpoint accepts plain bytes, not hex-encoded bytes.

File: lib/core/test/unit/Cardano/Wallet/Api/Malformed.hs [hunk 3]
```diff
@@ -113,7 +113,9 @@ import Servant
 
 import qualified Data.Aeson as Aeson
 import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BSL
```
Comment (rvl) [inline lib/core/test/unit/Cardano/Wallet/Api/Malformed.hs]:
```suggestion
import qualified Data.ByteString.Lazy as BL
```

**Review by rvl — APPROVED** [review]
Thanks.

File: lib/core/test/unit/Cardano/Wallet/Api/Malformed.hs (hunk 1 above)
Comment (sevanspowell) [reply lib/core/test/unit/Cardano/Wallet/Api/Malformed.hs]: Ah yes, you're right. That works!

**Comment by sevanspowell** [comment]
bors try

File: lib/core/test/unit/Cardano/Wallet/Api/Malformed.hs [hunk 4]
```diff
@@ -1513,7 +1513,11 @@ instance Malformed (BodyParam (ApiBytesT 'Base64 SerialisedTx))
 -- in Server.hs. Tested by integration tests.
 
 instance Malformed (BodyParam (ApiT SealedTx)) where
-    malformed = [] -- fixme: add a couple things
+    malformed = first BodyParam <$>
+        [ ("", "DecoderErrorDeserialiseFailure 'Shelley Tx' (DeserialiseFailure 0 'end of input')")
+        , ("hello", "DecoderErrorDeserialiseFailure 'Shelley Tx' (DeserialiseFailure 0 'expected list len or indef')")
+        , ("cafecafe", "DecoderErrorDeserialiseFailure 'Shelley Tx' (DeserialiseFailure 0 'expected list len or indef')")
```
Comment (rvl) [inline lib/core/test/unit/Cardano/Wallet/Api/Malformed.hs]: We probably only need one of these examples. They fail for the same reason.

**Review by rvl — APPROVED** [review]
Thanks, please squash before merging.

**Comment by sevanspowell** [comment]
bors retry

**Pull request merged**